apps/dalmatiner_db/src/dalmatiner_read_fsm.erl

```diff
@@ -111,9 +113,9 @@ init([ReqId, {VNode, System}, Op, From, Entity]) ->
     init([ReqId, {VNode, System}, Op, From, Entity, undefined, []]);
 
 init([ReqId, {VNode, System}, Op, From, Entity, Val, Opts]) ->
-    RR = not proplists:get_bool(no_rr, Opts),
-    {ok, N} = get_opt_or_env(dalmatiner_db, n, Opts),
-    {ok, R} = get_opt_or_env(dalmatiner_db, r, Opts),
+    {ok, RR} = get_rr_opt(Opts),
+    {ok, R} = get_r_opt(Opts),
+    {ok, N} = get_n_opt(Opts),
```

Comment:
I would swap them around so that `get_r_opt(Opts, N)` passes the `N` value instead of computing it again.

While the optimisation is minor it makes clearer that `R` also depends on `N`.